### PR TITLE
cli: don't check for updates to npm when we are updating npm itself

### PR DESCRIPTION
**Pull request by olore**
This is to resolve: https://npm.community/t/npm-warns-about-new-version-while-installing-new-npm-version/655

Skip checking for updates if we are updating npm [similar to how pnpm does it](https://github.com/pnpm/pnpm/blob/31595a918f61e1be16d225a260941f5c1b7c02e6/packages/pnpm/src/main.ts#L174)

### CODE REVIEW

File: bin/npm-cli.js [hunk 1]
```diff
@@ -75,6 +75,7 @@
   npm.load(conf, function (er) {
     if (er) return errorHandler(er)
     if (
+      !npm.argv.includes('npm') &&
```
Comment (zkat) [inline bin/npm-cli.js]: You'll note that [`pnpm` also makes sure it's a relevant command, as well as whether we're in `global` mode](https://github.com/pnpm/pnpm/blob/31595a918f61e1be16d225a260941f5c1b7c02e6/packages/pnpm/src/main.ts#L170). I think both of those check are worth doing.

I also agree with what pnpm is doing as far as disabling this on CI. The [`ci-info` package by @watson](https://www.npmjs.com/package/ci-info) seems to do a pretty good job at detecting this, so maybe that's worth pulling in (/cc @iarna).

**Review by zkat — CHANGES_REQUESTED** [review]
I think we need to do a bit more with this before it's mergeable. Thanks for taking the time to do this, though! It's super helpful to make sure this improves.

File: bin/npm-cli.js (hunk 1 above)
Comment (olore) [reply bin/npm-cli.js]: Thanks @zkat! I made updates to check install/update and if global mode.

Disregarding the CI stuff here based on comments on #33

**Comment by olore** [comment]
@zkat I think we're all set. Let me know if anything else.

**Review by zkat — APPROVED** [review]

**Comment by zkat** [comment]
Looks great! Thanks. 🎉

**Pull request merged**